client/app.js

```diff
@@ -1,56 +1,172 @@
 import React from "react";
+import "materialize-css/dist/css/materialize.min.css";
 import "./app.css";
-import { Route } from "react-router-dom";
+import { Route, Redirect } from "react-router-dom";
 //component
-import AllItems from "./Components/Items/AllItems";
 import DetailedItem from "./Components/Items/DetailedItem";
 import Home from "./Components/Home";
 import Login from "./Components/Users/LogIn";
 import NavBar from "./Components/NavBar";
 import Cart from "./Components/Cart/Cart";
-import Checkout from "./Components/Cart/Checkout";
-import "materialize-css/dist/css/materialize.min.css";
+import Dashboard from "./Components/Users/Dashboard";
+import SingleItemUpdate from "./Components/Items/SingleItemUpdate";
+import ItemCreate from "./Components/Items/ItemCreate";
+import UserUpdate from "./Components/Users/UserUpdate";
+import List from "./Components/List/List";
+import Footer from "./Components/Footer";
+import AdminDashboard from "./Components/Admin/Admin";
 import { verifyUserCookie } from "./Redux/User/actions/user.actions";
+import {
+  fetchOrderBySession,
+  fetchOrderStartAsync
+} from "./Redux/Order/actions/order.actions";
 import { connect } from "react-redux";
+import Admin from "./Components/Admin/Admin";
 
 export class App extends React.Component {
   componentDidMount() {
-   this.props.persistUser();
+    this.props.persistUser();
+    if (document.cookie) {
+      this.props.fetchOrder();
+    }
   }
+
+  componentDidUpdate(prevState, prevProps) {
+    if (prevState.user.loggedIn !== this.props.user.loggedIn) {
+      this.props.persistUser();
+    }
+  }
+
   render() {
     return (
       <div>
         <main>
           <NavBar />
+
+          <p className="User">
+            {!this.props.user.currentUser
+              ? "Loading"
+              : `Hey ${this.props.user.currentUser.firstName}`}
+          </p>
         </main>
-        <div class="col s12">
-          <Route exact path={"/"} render={() => <Home />} />
-          <Route exact path={"/shop"} render={() => <AllItems />} />
+        <div class="page-container">
+          <Route
+            exact
+            path={"/"}
+            render={() => <Home user={this.props.user} />}
+          />
+          <Route
+            exact
+            path={"/shop/pages/:pageId"}
+            render={() => (
+              <List
+                type="items"
+                pagination={true}
+                filterMethods={["size", "price", null]}
+                perPage={6}
+              />
+            )}
+          />
+          {/* START OF ADMIN ROUTES */}
+          <Route path={"/admin"} render={() => <AdminDashboard />}>
```

Comment:
can be overwritten in the DOM because this is on the client side, handle on the back end